django/utils/timezone.py

```diff
@@ -20,9 +28,10 @@
     'is_aware', 'is_naive', 'make_aware', 'make_naive',
 ]
 
-
+# ???: Can we have a fallback for this to pytz.utc?
```

Comment:
I might be missing something, but if it's just for UTC, could we not just use `datetime.timezone.utc` as a fallback?

Reply:
I think that the issue is that with `django.utils.timezone.utc` being `pytz.utc`, there is the expectation that the pytz API will be available. Just changing this to `datetime.timezone.utc` unconditionally is backward incompatible. Such a change was already reverted in the past in 2ec151e35da93047acfeea1b79c27010f2cb8594.

Moving forward, we want this to be `pytz.utc` if `USE_PYTZ_DEPRECATION_SHIM` is `True` else we want `datetime.timezone.utc`.

Reply:
The regression that led to reporting the change was:  `get_default_timezone()` returns a different object from `timezone.utc` when the default timezone is UTC. (A clear case of Hyrum's law...) See https://code.djangoproject.com/ticket/29514 for details.

@pope1ni's proposal looks like the correct way to preserve this invariant.

Reply:
As to whether it's "safe" to change from `pytz.UTC` to a standard `tzinfo` universally, I'd say that it's _not_ perfectly backwards compatible because `datetime.timezone.utc` lacks the `.localize` and `.normalize` methods that `pytz`'s time zones have. For UTC specifically, `pytz` doesn't _need_ these methods, since it's a fixed offset, but if someone has code assuming the datetimes are localized with `pytz` zones (sometimes UTC and sometimes not), they may be doing stuff like: `dt.tzinfo.normalize(dt + some_timedelta)`, which would start to fail in this case.

I could have sworn that there was some deprecation warning in earlier versions stating that fixed offsets and UTC would be transitioning to `datetime.timezone`, which would be a decent case for saying, "In the case of `UTC` we've had the deprecation warning in for long enough that you should have changed over anyway", but if you're already supporting a "fall back to `pytz`" option, it probably doesn't hurt to do the migration from `pytz.UTC` to `datetime.timezone.utc` under the same feature flag.